src/charm.py

```diff
@@ -30,105 +22,63 @@ class CephFSServerProxyCharm(ops.CharmBase):
-                "monitor_hosts": monitor_hosts,
-                "username": username,
-                "key": cephx_key,
-            },
-        )
-
-        self.unit.status = ops.ActiveStatus("Exporting share")
+        fsid = cast(str, config["fsid"])
 
-    def _on_share_requested(self, event: ShareRequestedEvent) -> None:
-        """Handle when CephFS client requests a CephFS share."""
-        logger.debug(("CephFS share requested with parameters " f"('name'={event.name})"))
-        if not (mount_info := self.mount_info):
-            logger.warning("Deferring ShareRequested event because filesystem info is not set")
-            self.unit.status = ops.BlockedStatus("No configured filesystem info")
-            event.defer()
-            return
-        share_info, auth_info = mount_info
-        logger.debug(f"CephFS mount info set to {share_info}")
-        self._cephfs_share.set_share(event.relation.id, share_info=share_info, auth_info=auth_info)
-
-    @property
-    def mount_info(self) -> Optional[Tuple[CephFSShareInfo, CephFSAuthInfo]]:
-        """Get the required info to mount the proxied CephFS share."""
-        if not (config := self.get_state("config")):
+        # Expected format: <filesystem_name>:<filesystem_shared_path>
+        sharepoint_cfg = cast(str, config["sharepoint"])
+        sharepoint = sharepoint_cfg.split(":", maxsplit=1)
+        if len(sharepoint) != 2 or not sharepoint[0] or not sharepoint[1]:
+            msg = f"Invalid sharepoint `{sharepoint_cfg}`"
+            logger.error(msg)
+            self.unit.status = ops.BlockedStatus(msg)
             return
+        name, path = sharepoint
 
-        share_info = CephFSShareInfo(
-            fsid=config["fsid"],
-            name=config["name"],
-            path=config["path"],
-            monitor_hosts=config["monitor_hosts"],
-        )
-        auth_info = CephFSAuthInfo(username=config["username"], key=config["key"])
-
-        return (share_info, auth_info)
-
-    @property
-    def peers(self) -> Optional[ops.Relation]:
-        """Fetch the peer relation."""
-        return self.model.get_relation(PEER_NAME)
-
-    def get_state(self, key: str) -> dict[Any, Any]:
-        """Get a value from the global state."""
-        if not self.peers:
-            return {}
+        # Expected format: <ip/hostname>:<port> <ip/hostname>:<port>
+        monitor_hosts = cast(str, config["monitor-hosts"]).split()
 
-        data = self.peers.data[self.app].get(key, "{}")
-        return json.loads(data)
+        # Expected format: <username>:<cephx-base64-key>
+        auth_info_cfg = cast(str, config["auth-info"])
+        auth_info = auth_info_cfg.split(":", maxsplit=1)
+        if len(auth_info) != 2 or not auth_info[0] or not auth_info[1]:
+            msg = f"Invalid auth-info `{auth_info_cfg}`"
+            logger.error(msg)
+            self.unit.status = ops.BlockedStatus(msg)
+            return
+        user, key = auth_info
```

Comment:
[nit]: I like to have one newline after the end of an if-else block. Not a huge issue though :sweat_smile: 

```suggestion

        user, key = auth_info
```